pkg/handler/v1/provider/openstack.go

```diff
@@ -45,17 +56,17 @@ func OpenstackSizeEndpoint(seedsGetter provider.SeedsGetter, presetsProvider pro
 			return nil, fmt.Errorf("error getting dc: %v", err)
 		}
 
-		username, password, domain, tenant, tenantID, err := getOpenstackCredentials(userInfo, req.Credential, req.Username, req.Password, req.Domain, req.Tenant, req.TenantID, presetsProvider)
+		credential, err := getCredentials(userInfo, req, presetsProvider)
 		if err != nil {
-			return nil, fmt.Errorf("error getting OpenStack credentials: %v", err)
+			return nil, err
 		}
 
 		settings, err := settingsProvider.GetGlobalSettings()
 		if err != nil {
 			return nil, common.KubernetesErrorToHTTPError(err)
 		}
 
-		return providercommon.GetOpenstackSizes(username, password, tenant, tenantID, domain, datacenterName, datacenter, settings.Spec.MachineDeploymentVMResourceQuota)
+		return providercommon.GetOpenstackSizes(credential.Username, credential.Password, credential.Tenant, credential.TenantID, credential.Domain, datacenterName, datacenter, settings.Spec.MachineDeploymentVMResourceQuota)
```

Comment:
You could also consider changing the signature of the function to pass credential object directly e.g.
```
providercommon.GetOpenstackSizes(credential, datacenterName, settings.Spec.MachineDeploymentVMResourceQuota)
```

Reply:
Since this is a public method. I didn't change it in order to avoid any breaking in other part of the code where these functions are being called.

Reply:
This is a very good point @imharshita, note that you could address this by using a different function name as Goland does not support function overloading. e.g.

```
func GetOpenstackSizesWithTenantReq(... ) (...) {
    return GetOpenstackSizes(...)
}
```

Secondly, we are not as strict as we should be with API compatibility in Kubermatic repo as it is not supposed to be imported as a library. But still your attention to this is very much appreciated ;-)

Reply:
Regarding this comment, I can surely do what you suggested, but that will require us to apply the same 5-6 methods.

Reply:
don't bother, it's not a blocker